01_prepare_host.sh

```diff
@@ -170,10 +170,20 @@ if [ ! -f "${IMAGE_NAME}" ] ; then
 fi
 popd
 
+# If a given container with tag doesn't exist locally, pull it. 
+# Otherwise, do nothing.
+# Helps conserve number of API calls to DockerHub to avoid hitting rate limit.
+function container_image_pull_if_missing() {
+  local IMAGE="$1"
+  if ! sudo "${CONTAINER_RUNTIME}" | grep -q "$IMAGE"; then
+    sudo "${CONTAINER_RUNTIME}" pull "$IMAGE"
+  fi
+}
+
```

Comment:
I am thinking shall we keep this function in lib/common.sh, if other scripts might need it? Not sure if they need it though.

Reply:
I've moved the function to lib/common.sh.

Reply:
Pulling image is used only in 01 script.